### PR TITLE
chore: init 2.0

### PR DESCRIPTION
**Pull request by gberenice**
## Info
* Updates the module template with:
  * trunk
  * aqua
  * `.editorconfig`
  * `.gitignore`
  * GHA: lint workflow
  * pre-commit (for terraform docs that is not currently supported by trunk)
  * `context.tf`
 * Assuming that community health files will be added to the org `.github` repo.

### CODE REVIEW

**Review by Gowiem — DISMISSED** [review]
Couple small comments. Let's change the markdown trailing whitespace issue if you don't know why we have it either, but otherwise :shipit: !

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.{tf,tfvars}]
+indent_size = 2
+indent_style = space
+
+[*.md]
+max_line_length = 0
+trim_trailing_whitespace = false
```
[the remark this answers is not in the document]
Comment (gberenice) [reply .editorconfig]: Good call. I think this should have been fixed.

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Masterpoint
+Copyright (c) 2024 Masterpoint
```
[the remark this answers is not in the document]
Comment (gberenice) [reply LICENSE]: Replaced with Apache 2.0. I should read more into this stuff 🤔

**Review by Gowiem — APPROVED** [review]
:shipit: @gberenice -- well done!

**Pull request merged**